src/overlay005/field_control.c

```diff
@@ -0,0 +1,1087 @@
+            int event = sub_0203CA6C(fieldSystem, sub_0203A440(fieldSystem), sub_0203A448(fieldSystem));
+
+            if (event != 0xffff) {
+                sub_0203E880(fieldSystem, event, NULL);
+                return TRUE;
+            }
+        }
+
+        if (distortionState == AVATAR_DISTORTION_STATE_NONE) {
+            int event;
+
+            if (ov5_021EFB40(fieldSystem, &event)) {
+                sub_0203E880(fieldSystem, event, NULL);
+                return TRUE;
+            }
+        }
+
+        int distortionDir = sub_0205EAA0(fieldSystem->playerAvatar);
+        u32 distortionBehavior = sub_020616F0(fieldSystem->playerAvatar, distortionDir);
+        int distortionScript = Field_TileBehaviorToScript(fieldSystem, distortionBehavior);
+
+        if (distortionScript != 0xffff) {
+            sub_0203E880(fieldSystem, distortionScript, NULL);
+            return TRUE;
+        }
+
+        int playerX, playerZ;
+        int playerDir = PlayerAvatar_GetDir(fieldSystem->playerAvatar);
+
+        Field_Step(fieldSystem, &playerX, &playerZ);
+
+        if (ov5_021F8410(fieldSystem, playerX, playerZ, playerDir) == TRUE) {
+            return TRUE;
+        }
+    }
+
+    if (input->mapTransition) {
+        if (sub_02071CB4(fieldSystem, 9) == TRUE) {
+            ov9_0224A800(fieldSystem, input->transitionDir);
+        } else if (Field_CheckMapTransition(fieldSystem, input) == TRUE) {
+            Field_TrySetMapConnection(fieldSystem);
+            return TRUE;
+        }
+    }
+
+    if (input->registeredItem && sub_02069238(fieldSystem) == TRUE) {
+        return TRUE;
+    }
+
+    if (input->menu && sub_0203A9C8(fieldSystem) == TRUE) {
+        Sound_PlayEffect(SEQ_SE_DP_WIN_OPEN);
+        FieldMenu_Init(fieldSystem);
+        return TRUE;
+    }
+
+    return FALSE;
+}
+
+static BOOL Field_CheckSign (FieldSystem *fieldSystem)
+{
+    MapObject *object;
+
+    if (sub_0203CBE0(fieldSystem, &object) == TRUE) {
+        sub_0203E880(fieldSystem, sub_02062960(object), object);
+        return TRUE;
+    }
+
+    int event = sub_0203CB80(fieldSystem, sub_0203A440(fieldSystem), sub_0203A448(fieldSystem));
+    
+    if (event != 0xffff) {
+        sub_0203E880(fieldSystem, event, NULL);
+        return TRUE;
+    }
+
+    return FALSE;
+}
+
+BOOL FieldInput_Process_Underground (FieldInput *input, FieldSystem *fieldSystem)
+{
+    if (input->dummy5 == FALSE && sub_0203F5C0(fieldSystem, 1) == TRUE) {
+        return TRUE;
+    }
+
+    if (sub_020363A0() || CommPlayer_GetMovementTimer(CommSys_CurNetId()) != 0) {
+        return FALSE;
+    }
+
+    ov23_02242814();
+
+    if (input->interact) {
+        ov23_02242830(FALSE);
+        return FALSE;
+    }
+
+    if (input->menu) {
+        ov23_022427F8();
+        return FALSE;
+    }
+
+    if (ov23_02242458()) {
+        return FALSE;
+    }
+
+    return FALSE;
+}
+
+BOOL FieldInput_Process_Colosseum (FieldInput *input, FieldSystem *fieldSystem)
```

Comment:
I'm realizing now that I don't understand what this is referring to. What's the Colosseum here?

Reply:
The battle colosseum on the top floor of pokemon centers, next to the union room. I initially forgot that was a thing too lol

Reply:
I have no idea what this feature is... Do you mean the usual connectivity options using the native DS wireless features? (i.e., not WiFi, which is in the basement)